src/gsy_e/models/area/scm_manager.py

```diff
@@ -445,12 +472,24 @@ def calculate_home_energy_bills(
 
         self._bills[home_uuid] = home_bill
 
+    def accumulate_community_trades(self):
+        """
+        Gather all trades from homes to the community after meter data, in order to have them
+        available for the simulation results generation.
+        """
+        for home_data in self._home_data.values():
+            self.community_data.trades.extend(home_data.trades)
+
     def get_area_results(self, area_uuid: str, serializable: bool = False) -> Dict:
         """Return the SCM results for one area (and one time slot)."""
         if area_uuid == self._community_uuid:
             return {
                 "bills": self.community_bills,
-                "after_meter_data": {}
+                "after_meter_data": self.community_data.serializable_dict(),
+                "trades": [
+                    trade.serializable_dict()
+                    for trade in self.community_data.trades
+                ]
```

Comment:
Is this needed? Isn't it just a duplicate of `["after_meter_data"]["trades"]`?

Reply:
It is. However the community area needs to have the community trades in order to generate the results. Keep in mind that all results handlers operate on individual areas / markets, therefore if there are no trades in the community then simply no results will be generated. This is the actual root cause of the "no results are generated for communities" and the main fix here

Reply:
I'm not sure I understood correctly. As far as I see, all "area results" (for all areas) will contain an `after_meter_data` key. Is this correct? 
AFAIU, `area["after_meter_data"]["trades"]` will be present for community areas, even if we don't duplicate it in `area["trades"]`. Could you please share a link of where this would break?

Reply:
https://github.com/gridsingularity/gsy-framework/blob/master/gsy_framework/sim_results/all_results.py#L68 have a look at this method. This iterates over all areas, passes the trade objects associated with each area to each simulation result handler, and calculates the results / updates each handler according to the trades that are associated with each area. If the community area does not have associated trades, then its handlers will not be updated because they assume that no trades happened for this area. 
Keep in mind, that this needs to be like this in order to support our "market" model, where the trades from different areas are represented as chain trades. Therefore in order to reuse this logic, we have to duplicate the home trades, and simulate that a "chain trade" happened on the community level too. That way the calculations are only per-area and decoupled. 
Let me know if this is not clear enough